src/quo2/components/dropdown.cljs

```diff
@@ -0,0 +1,150 @@
+(ns quo2.components.dropdown
+  (:require [quo.react-native :as rn]
+            [quo2.components.icon :as icons]
+            [quo2.components.text :as text]
+            [quo2.foundations.colors :as colors]
+            [quo2.reanimated :as reanimated]
+            [reagent.core :as reagent]))
+
+(defn apply-anim [dd-height val]
+  (reanimated/animate-shared-value-with-delay dd-height
+                                              val
+                                              300
+                                              :easing1
+                                              0))
+
+(def sizes {:big {:icon-size 20
+                  :font {:font-size :paragraph-1}
+                  :height 40
+                  :padding {:padding-with-icon {:padding-top 9
+                                                :padding-bottom 9
+                                                :padding-left 12
+                                                :padding-right 12}
+                            :padding-with-no-icon {:padding-top 9
+                                                   :padding-bottom 9
+                                                   :padding-left 16
+                                                   :padding-right 12}}}
+            :medium {:icon-size 20
+                     :font {:font-size :paragraph-1}
+                     :height 32
+                     :padding {:padding-with-icon {:padding-top 5
+                                                   :padding-bottom 5
+                                                   :padding-horizontal 8}
+                               :padding-with-no-icon {:padding-top 5
+                                                      :padding-bottom 5
+                                                      :padding-left 12
+                                                      :padding-right 8}}}
+            :small {:icon-size 12
+                    :font {:font-size :label}
+                    :height 24
+                    :padding {:padding-with-icon {:padding-vertical 3
+                                                  :padding-left 6
+                                                  :padding-right 6}
+                              :padding-with-no-icon {:padding-vertical 3
+                                                     :padding-horizontal 8}}}})
+(defn color-by-10
+  [color]
+  (colors/alpha color 0.6))
+
+(defn dropdown-comp [{:keys [icon open? text dd-height size disabled? dd-color border-color]}]
+  (let [dark? (colors/dark?)
+        {:keys [width height width-with-icon padding font icon-size]} (get-in sizes [size])
```

Comment:
The `get-in` functions doesn't seem to be necessary. It's also common in Clojure to treat maps as functions when there's a guarantee they're never nil, which is normally the case for maps defined as vars, like `sizes`. So if you prefer, the refactored code could be:

```clojure
{:keys [width height width-with-icon padding font icon-size]} (sizes size)
```

Reply:
Thanks .. Used it